src/docfx/restore/RestoreGit.cs

```diff
@@ -42,162 +45,66 @@ public static async Task<IEnumerable<(string href, string workTreeHead)>> Restor
-                    {
-                        await GitUtility.AddWorkTree(restorePath, workTreeHead, workTreePath);
-                    }
-
-                    workTreeHeads.Add((href, workTreeHead));
-                });
-            }
-
-            Debug.Assert(hrefs.Count == workTreeHeads.Count);
-            return workTreeHeads.ToList();
-        }
-
-        // clean up un-used work trees
-        private static async Task<List<string>> CleanupWorkTrees(string restoreDir)
-        {
-            var remainingWorkTrees = new List<string>();
-
-            if (!GitUtility.IsRepo(restoreDir))
-            {
-                return remainingWorkTrees;
-            }
-
-            var restorePath = PathUtility.NormalizeFolder(Path.Combine(restoreDir, ".git"));
+                    return default;
+                }
 
-            await ProcessUtility.RunInsideMutex(
-                PathUtility.NormalizeFile(Path.GetRelativePath(AppData.GitRestoreDir, restorePath)),
-                async () =>
+                if (config.LocMappingType != LocMappingType.Repository && config.LocMappingType != LocMappingType.RepositoryAndFolder)
                 {
-                    var existingWorkTrees = await GitUtility.ListWorkTrees(restorePath, false);
-                    if (NeedCleanupWorkTrees(existingWorkTrees.Count))
-                    {
-                        remainingWorkTrees = await CleanupWorkTrees(existingWorkTrees);
-                    }
-                });
-
-            return remainingWorkTrees;
-
-            bool NeedCleanupWorkTrees(int existingWorkTreeCount) => existingWorkTreeCount > MaxWorkTreeCount;
+                    return default;
+                }
 
-            async Task<List<string>> CleanupWorkTrees(List<string> existingWorkTrees)
-            {
-                var allWorkTreesInUse = await GetAllWorkTreePaths(restoreDir);
-                var leftWorkTrees = new List<string>();
-                foreach (var workTree in existingWorkTrees)
+                var repo = Repository.Create(Path.GetFullPath(docsetPath));
+                if (repo == null)
                 {
-                    if (!allWorkTreesInUse.Contains(workTree))
-                    {
-                        // remove not used work tree
-                        Directory.Delete(workTree, true);
-                    }
-                    else
-                    {
-                        leftWorkTrees.Add(workTree);
-                    }
+                    return default;
                 }
-                await GitUtility.PruneWorkTrees(restorePath);
 
-                return leftWorkTrees;
+                var locRepoFullName = LocConfigConvention.GetLocRepository(config.LocMappingType, repo.FullName, locale, config.DefaultLocale);
+                var locRepo = repo.With(repo.Owner, locRepoFullName.Split(new[] { '\\', '/' }, StringSplitOptions.RemoveEmptyEntries).Last());
```

Comment:
>repo.Owner [](start = 40, length = 10)

Updating with the same owner? Seems only need to set Name here, do we need a With method?